workers/data_refinery_workers/processors/management/commands/correct_affy_samples.py

```diff
@@ -0,0 +1,145 @@
+"""Tries to find and correct mislabelled Affymetrix samples.
+
+See https://github.com/AlexsLemonade/refinebio/issues/968 for more
+context, but essentially we had a bug in the surveyor where if a GEO
+superseries had an RNA-Seq sample in it, then all the samples would be
+labelled as RNA-SEQ, including samples that were actually
+Microarray. This command will correct any such Affymetrix samples by
+checking their filenames to see if they are .CEL files. If they are
+the file is downloaded and we use affy.io to read the header and
+determine the platform accesion code and name.
+"""
+
+import os
+import shutil
+import string
+import urllib
+from typing import Dict
+
+from django.core.management.base import BaseCommand
+from rpy2.rinterface import RRuntimeError
+import rpy2.robjects as ro
+
+from data_refinery_common.logging import get_and_configure_logger
+from data_refinery_common.utils import (
+    get_env_variable,
+    get_readable_affymetrix_names,
+)
+from data_refinery_common.models import *
+
+
+logger = get_and_configure_logger(__name__)
+CHUNK_SIZE = 1024 * 256 # chunk_size is in bytes
+
+
+def _download_file(download_url: str, file_path: str) -> None:
+    """Download a file from GEO.
+    """
+
+    try:
+        logger.debug("Downloading file from %s to %s.",
+                     download_url,
+                     file_path)
+
+        # Ancient unresolved bug. WTF python: https://bugs.python.org/issue27973
+        urllib.request.urlcleanup()
+
+        target_file = open(file_path, "wb")
+        with urllib.request.urlopen(download_url) as request:
+            shutil.copyfileobj(request, target_file, CHUNK_SIZE)
+
+        urllib.request.urlcleanup()
+    except Exception:
+        logger.exception("Exception caught while downloading file %s from %s",
+                         file_path, download_url)
+        raise
```

Comment:
In general I find this pattern to be a bit weird (I'd rather it be do a thing vs fail or return True/False rather than True/raise, this seems to be both) but it looks like you're handling it okay and this function is only used here so it's not a big deal. I'd refactor if you've got spare cycles but if you don't it's NBD.